### PR TITLE
Drawer: Add tests

### LINKED ISSUE #362: Drawer: Add tests

### PR DESCRIPTION
**Pull request by westwood846**
Closes #362

### CODE REVIEW

File: src/Navigation/Drawer/Drawer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+
+import * as renderer from 'react-test-renderer';
+import '@testing-library/jest-dom/extend-expect';
+import {
+  fireEvent,
+  render,
+  queryByTestId,
+  queryByRole,
```
Comment (rosslo) [inline src/Navigation/Drawer/Drawer.test.tsx]: eslint warning: `'queryByRole' is defined but never used`

File: src/Navigation/Drawer/Drawer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+} from '@testing-library/react';
+
+import Drawer from './Drawer';
+
+describe('<Drawer /> render', () => {
+  test('should match snapshot when open', () => {
+    const snapshot = renderer
+      .create(
+        <Drawer isOpen={false} onClose={jest.fn()}>
```
Comment (rosslo) [inline src/Navigation/Drawer/Drawer.test.tsx]: According to the test name, I think the `isOpen` should be `true`

File: src/Navigation/Drawer/Drawer.test.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    render(
+      <Drawer isOpen={false} onClose={jest.fn()}>
+        test-content
+      </Drawer>
+    );
+
+    const drawerWrapper = queryByTestId(document.body, 'drawer-wrapper');
+    fireEvent.click(drawerWrapper);
+
+    expect(windowClickEventListener).not.toHaveBeenCalled();
```
Comment (rosslo) [inline src/Navigation/Drawer/Drawer.test.tsx]: I think we can just check `expect(onClose).not.toHaveBeenCalled();` instead.

File: src/Navigation/Drawer/Drawer.test.tsx (hunk 1 above)
Comment (westwood846) [reply src/Navigation/Drawer/Drawer.test.tsx]: Removed

File: src/Navigation/Drawer/Drawer.test.tsx (hunk 2 above)
Comment (westwood846) [reply src/Navigation/Drawer/Drawer.test.tsx]: Thanks for the catch!

File: src/Navigation/Drawer/Drawer.test.tsx (hunk 3 above)
Comment (westwood846) [reply src/Navigation/Drawer/Drawer.test.tsx]: Oh now I get what that is for! ^^ Changed it.

**Review by rosslo — APPROVED** [review]

**Pull request merged**